### ISSUE TITLE
Nuget pack from csproj gives Unable to find 'CmdLine.1.0.7.509.nupkg'. 

### ISSUE THREAD

**Issue by kingerja** [NuGet/Home#3198]
This only fails in my lab TFS builds.  It works fine locally.  I am currently working on creating a repro to see if I can isolate it further.  

I am using the latest dev build: NuGet.exe 3.5.0.1646

see attached zip file to look at the csproj/nuspec/packages.config.  (Note that is not a contained repro so you will not be able to build it.  I will update this issue when I get an isolated repro.)

```
   NuGetExecPack:
```

```
         IF NOT EXIST "F:\_Bld\12657\750\Binaries\AnyCPU\Release\logs" MD "F:\_Bld\12657\750\Binaries\AnyCPU\Release\logs"
         SET VSS_NUGET_VERBOSITY=Detailed
         SET VSS_NUGET_LOGFILEPREFIX=F:\_Bld\12657\750\Binaries\AnyCPU\Release\logs\credentialprovider.20160722.GatedBalance.csproj
         REM Nuget execution time 7/22/2016 8:19:40 PM

         SET NUGET_PACKAGES=F:\nugetRoot
         IF NOT DEFINED SkipNugetTries SET NugetTries=5
         SET NugetTryDelay=30
         SET NugetTMP=F:\_Bld\12657\750\Sources\obj\AnyCPU\Release\GatedBalance.csproj\
         "F:\_Bld\12657\750\Sources\CBT\NuGet\RetriableNuGet.cmd" pack "GatedBalance.csproj" -p ";ModulePath=tools\Modules;Configuration=Release;TeamBuildBaseOutputPath=F:\_Bld\12657\750\Binaries\AnyCPU\Release;BuildingInTeamBuild=true;BuildingInLab=true;Platform=AnyCPU;EnlistmentRoot=F:\_Bld\12657\750\Sources;packageversion=1.1.3394.1;versionmajor=1;versionminor=1;versionbuildnumber=3394;versionbuildrevision=1" -o "F:\_Bld\12657\750\Binaries\AnyCPU\Release\Product\GatedBalance" -BasePath "F:\_Bld\12657\750\Binaries\AnyCPU\Release\Product\GatedBalance" -Exclude *.nupkg -NonInteractive -NoPackageAnalysis -Tool
    94>NuGetExecPack:
         Attempting to build package from 'GatedBalance.csproj'.
         MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'.
         Packing files from 'F:\_Bld\12657\750\Binaries\AnyCPU\Release\Product\GatedBalance'.
         Using 'GatedBalance.nuspec' for metadata.
         Found packages.config. Using packages listed as dependencies
    94>F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error : Unable to find 'CmdLine.1.0.7.509.nupkg'. Make sure the project has been built. [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
    94>F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: The command " [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073:          [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073:          [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: IF NOT EXIST "F:\_Bld\12657\750\Binaries\AnyCPU\Release\logs" MD "F:\_Bld\12657\750\Binaries\AnyCPU\Release\logs" [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: SET VSS_NUGET_VERBOSITY=Detailed [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: SET VSS_NUGET_LOGFILEPREFIX=F:\_Bld\12657\750\Binaries\AnyCPU\Release\logs\credentialprovider.20160722.GatedBalance.csproj [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: REM Nuget execution time 7/22/2016 8:19:40 PM [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073:  [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: SET NUGET_PACKAGES=F:\nugetRoot [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: IF NOT DEFINED SkipNugetTries SET NugetTries=5 [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: SET NugetTryDelay=30 [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: SET NugetTMP=F:\_Bld\12657\750\Sources\obj\AnyCPU\Release\GatedBalance.csproj\ [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073: "F:\_Bld\12657\750\Sources\CBT\NuGet\RetriableNuGet.cmd" pack "GatedBalance.csproj" -p ";ModulePath=tools\Modules;Configuration=Release;TeamBuildBaseOutputPath=F:\_Bld\12657\750\Binaries\AnyCPU\Release;BuildingInTeamBuild=true;BuildingInLab=true;Platform=AnyCPU;EnlistmentRoot=F:\_Bld\12657\750\Sources;packageversion=1.1.3394.1;versionmajor=1;versionminor=1;versionbuildnumber=3394;versionbuildrevision=1" -o "F:\_Bld\12657\750\Binaries\AnyCPU\Release\Product\GatedBalance" -BasePath "F:\_Bld\12657\750\Binaries\AnyCPU\Release\Product\GatedBalance" -Exclude *.nupkg -NonInteractive -NoPackageAnalysis -Tool [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
F:\_Bld\12657\750\Sources\CBT\NuGet\NuGet.targets(243,5): error MSB3073:       " exited with code 1. [F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj]
    94>Done Building Project "F:\_Bld\12657\750\Sources\Product\Source\GatedBalance\GatedBalance.csproj" (default targets) -- FAILED.

```

[GatedBalance.zip](https://github.com/NuGet/Home/files/379186/GatedBalance.zip)

**Comment by rrelyea** [comment]
Any progress on a narrowed repro?
Did this work with 3.5.0-beta2?

**Comment by kingerja** [comment]
I just tried it on 3.5.0-beta2 (3.5.0.1520) and it also contains this failure.
However 3.4.4 does not contain this issue.

I have attempted many repro's out side of our TFS lab env and have been unsuccessful.  In my lab it is a consistent repro.  However just running it on the build agent as the same command doesn't appear to repro it which leads me to suspect it is something to do with running under TFS.  

I created a stripped repo that does not depend on any build env.  If you run the repro.cmd in the context of a msbuild project (such as using an <Exec Command="Repro.cmd"/> task) and have your msbuild project run as a tfs lab build.  It should repro.
\joshking-main\dump\nugetissue\3198

If you don't have a TFS instance stood up with agents and such you can use and debug, have your dev contact me (joshking) and I can set them up with everything they need and walk them through how to queue a build and repro it.  (our build agents also have VS installed so they could debug from there.)

**Comment by kingerja** [comment]
Local repro found.  
Make sure you have a clean nuget cache then do the below:

robocopy \joshking-main\dump\nuget\issues\3198 d:\temp\nuget\issue\3198
cd /d d:\temp\nuget\issue\3198
nuget.exe pack GatedBalance.csproj

*note you may hit issue https://github.com/NuGet/Home/issues/3343 if you nuget packages dir doesn't exist yet.

**Comment by zhili1208** [comment]
@kingerja looks like nuget 3.4 behavior is a bug, 3.5 fixed it. NuGet needs dependency package for pack, so if it can't find that package, in 3.4, it will just not add that package as dependency, in 3.5, it will throw.
I think throw is right here, otherwise user got wrong package without any error.

**Comment by kingerja** [comment]
Thanks Zhi for helping me narrow down on an accurate repro and identifying the issue.
\joshking-main\dump\nuget\issues\3198\repro.cmd
The issue is that in the lab we install our packages using nuget restore -PackagesDirectory %enlistmentRoot%\external\packages
For our local dev builds we install the packages via the nuget.config configuration option which is set to \NugetPackages.
Nuget pack only considers the nuget.config for packagedirectory lookup and doesn't provide a cmdline override.

Apparently in an older 3.4 build this behavior was changed to validate the nuget packages were installed but it didn't fail with an error.  Now an error is being generated.  

I think the best approach is that if nuget pack is going to have a dependency on the packagedirectory then there should be an option to override that configuration via cmd line so perhaps adding -PackageDirectory as an option to the nuget pack command.

**Comment by zhili1208** [comment]
First I thought we can override nuget.config , but overriding nuget.config is disallowed for pack since nuget 2.8. So the best way here is adding -PackageDirectory as an option. filed a feature request bug for it.https://github.com/NuGet/Home/issues/3378

**Comment by zhili1208** [comment]
Since this is not a bug, close this one. Will fix this issue in #3378
